php/php.editor/src/org/netbeans/modules/php/editor/actions/FixUsesPerformer.java

```diff
@@ -631,13 +757,221 @@ private int getOffsetWithoutLeadingWhitespaces(final int startOffset) {
         return result;
     }
 
+    private int getOffsetWithoutFollowingWhitespaces(final int endOffset) {
+        int result = endOffset;
+        baseDocument.readLock();
+        try {
+            TokenSequence<PHPTokenId> ts = LexUtilities.getPHPTokenSequence(baseDocument, endOffset);
+            if (ts != null) {
+                ts.move(endOffset);
+                while (ts.moveNext() && ts.token().id().equals(PHPTokenId.WHITESPACE)) {
+                    result = ts.offset() + ts.token().length();
+                }
+            }
+        } finally {
+            baseDocument.readUnlock();
+        }
+        return result;
+    }
+
+    private class UsesInsertStringHelper {
+        private final Map<Integer, OffsetRange> usedRanges;
+        private final boolean appendUses;
+        private final boolean hasMultipleUses;
+        private StringBuilder resultString = new StringBuilder();
+        private Set<Integer> followingWhitespaceReplaceOffset = new HashSet<>();
+        private Set<Integer> savedHeadingWhitespaceOffset = new HashSet<>();
```

Comment:
`final`

Reply:
Yes, please, make all the fields `final`, if possible (and call `.clear()` and similarly, if needed).